assignment3.txt

```diff
@@ -0,0 +1,9 @@
+Julia Tan
+Assignment3 Part 2
+
+1. Yes, however it must be assumed that the variable 'i' has been initialized already.
+2. The code results in an infinite loop, because i decrements and therefore will always be less than 10.
+3.Determining complexity
+i. n^2
+ii.n^3
```

Comment:
The answer is actually O(n^4) - again, for loops that run from 0 to n have the time complexity O(n). So the top loop is O(n), the middle Loop is O(n^2) iterations because it iterates from 1 to i^2. The if statement is O(1 / n) times because of the modulo function. Lastly, the inner loop  is O(n^2) iterations because it iterates from 0 to j and j = i^2 = n^2. 

Therefore, the sum = -(1/12)*n + (3/8)*n^2 - (5/12)*n^3 + (1/8)*n^4  = O(n^4)